### PR TITLE
layers: Label Maintenance1 ImageView VU

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Part of #2945

Labels
- `VUID-VkImageViewCreateInfo-image-04970`
- `VUID-VkImageViewCreateInfo-image-04971` (add some test as well)

Also improved some of the surrounding error messages as well

### CODE REVIEW

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 1]
```diff
@@ -9328,6 +9328,55 @@ TEST_F(VkLayerTest, CreateImageViewInvalidSubresourceRange) {
                 m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-VkImageViewCreateInfo-image-02724");
                 CreateImageViewTest(*this, &img_view_info, "VUID-VkImageViewCreateInfo-subresourceRange-02725");
             }
+
+            // Checking sparse flags are not set
+            VkImageCreateInfo sparse_image_ci = image_ci;
+            sparse_image_ci.mipLevels = 1;
```
Comment (ncesario-lunarg) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:9334]: Looks like this is never used?

**Review by ncesario-lunarg — COMMENTED** [review]
Other than the unused variable (causing a warning on one of our linux build machines), LGTM!

File: tests/vklayertests_buffer_image_memory_sampler.cpp (hunk 1 above)
Comment (sfricke-samsung) [reply tests/vklayertests_buffer_image_memory_sampler.cpp]: ... ya guess not, I thought I would bring the miplevel to 1, but seems it didn't matter, removed

...Why did I not get the warning though 😕

**Review by ncesario-lunarg — APPROVED** [review]

File: tests/vklayertests_buffer_image_memory_sampler.cpp (hunk 1 above)
Comment (ncesario-lunarg) [reply tests/vklayertests_buffer_image_memory_sampler.cpp]: What platform/compiler are you using?

**Pull request merged**